org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java

```diff
@@ -51,7 +52,8 @@ public ServiceConfiguration(ServiceConfigurationKey<S> key) {
 	public ServiceConfiguration(Class<S> serviceType, String format, String[] args, int cardinality, long timeout) {
 		this.serviceType = requireNonNull(serviceType);
 
-		Filter filter = format("(objectClass=%s)", serviceType.getName());
+		String name= WildcardType.class.equals(serviceType) ? "*": serviceType.getName();
```

Comment:
If we are using AnyService, then we should _not_ put an objectClass expression in the filter. `(objectClass=*)` is just wasteful. It always evaluates to true for a service since all services have an objectClass property.